src/features/auth/signup/SetBirthday/utils/useDatePickerErrorHandler.tsx

```diff
@@ -0,0 +1,53 @@
+import { t } from '@lingui/macro'
+import { useEffect, useState } from 'react'
+
+import { useAppSettings } from 'features/auth/settings'
+import { analytics } from 'libs/analytics'
+import { dateDiffInFullYears } from 'libs/dates'
+import { formatDateToISOStringWithoutTime } from 'libs/parsers'
+
+const DEFAULT_YOUNGEST_AGE = 15
+const UNDER_YOUNGEST_AGE = DEFAULT_YOUNGEST_AGE - 1
+
+export const useDatePickerErrorHandler = (date?: Date) => {
+  const CURRENT_DATE = new Date()
+  const DEFAULT_SELECTED_DATE = new Date(
+    new Date().setFullYear(new Date().getFullYear() - UNDER_YOUNGEST_AGE)
+  )
+  const DEFAULT_SELECTED_DATE_WITHOUT_TIME = formatDateToISOStringWithoutTime(DEFAULT_SELECTED_DATE)
+
+  const [errorMessage, setErrorMessage] = useState<string | null>(null)
+  const [isDisabled, setIsDisabled] = useState(true)
+
+  const { data: settings } = useAppSettings()
+  const youngestAge = settings?.accountCreationMinimumAge ?? DEFAULT_YOUNGEST_AGE
+
+  useEffect(() => {
```

Comment:
t'as besoin d'un `useState` et d'un `useEffect` ?

Reply:
Lorsque je retire le useEffect j'ai un problème d'infinite loop 
`Too many re-renders. React limits the number of renders to prevent an infinite loop.`

Reply:
enlève le `useState` aussi tu verras ce sera mieux. En gros si t'avais pas besoin de `useAppSettings`, ce serait même pas un hook cette fonction.

Reply:
en plus ça devrait enlever qqs code smells détectés par sonarcloud